pkg/utils/parse_arf_result.go

```diff
@@ -453,10 +484,66 @@ func remediationsFromString(scheme *runtime.Scheme, name string, namespace strin
 	return rems
 }
 
+func getQuotedStrings(s string) []string {
+	matchStr := regexp.MustCompile(`".*?"`).FindAllString(s, -1)
+	matchList := make([]string, len(matchStr))
+	for i, m := range matchStr {
+		matchList[i] = m[1 : len(m)-1]
+	}
+	return matchList
+}
+
+func parseValues(remContent string, resultValues map[string]string) (string, []string, []string, error) {
+
+	t, err := template.New("").Option("missingkey=error").Parse(remContent)
+	var valuesUsedList []string
+	var valuesMissingList []string
+	if err != nil {
+		return remContent, valuesUsedList, valuesMissingList, err //Error creating template
+	}
+	buf := &bytes.Buffer{}
+
+	errMsg := t.Execute(buf, resultValues) //Will return error if a value is not found in XCCDF
+	fixedText := buf.String()
+	if errMsg == nil {
+		valuesUsedList = getParsedValueName(t)
+		return fixedText, valuesUsedList, valuesMissingList, errMsg
+	} else if strings.Contains(errMsg.Error(), "map has no entry") {
```

Comment:
this makes me a bit nervous..are there no error types returned from the template engine? Depending on a regex returned seems especially brittle..
In `getParsedValueName` you iterate the tree, would it be possible to get the missing values this way, too?

Reply:
Actually, I was going to implement an error message, but I haven't decided what to put, as right now, if there is an error during any time at parsing, it will return nil and skip creating the remediaton,

Reply:
The `getParsedValueName` tree is before its execution, the the parsing will stop upon Notfound value, so the parsing tree will have all the values to be parsed.

Reply:
I think I found a way around it, I am going to compare the Parsing values list with the found-value list to get the missing value list